ts/state/ducks/conversations.ts

```diff
@@ -3847,11 +3848,86 @@ function saveAttachment(
   };
 }
 
+const showSaveMultiDialog = (): Promise<{
+  canceled: boolean;
+  dirPath?: string;
+}> => {
+  return ipcRenderer.invoke('show-save-multi-dialog');
+};
+
+export type SaveAttachmentsActionCreatorType = ReadonlyDeep<
+  (
+    attachments: readonly AttachmentType[],
+    timestamp?: number,
+    index?: number
+  ) => unknown
+>;
+
+function saveAttachments(
+  attachments: readonly AttachmentType[],
+  timestamp = Date.now(),
+  index = 0
+): ThunkAction<void, RootStateType, unknown, ShowToastActionType> {
+  return async dispatch => {
+    // check if any of the attachments could be dangerous
+    for (const attachment of attachments) {
+      const { fileName = '' } = attachment;
+
+      const isDangerous = isFileDangerous(fileName);
+      if (isDangerous) {
+        dispatch({
+          type: SHOW_TOAST,
+          payload: {
+            toastType: ToastType.DangerousFileType,
+          },
+        });
+        return;
+      }
+    }
+
+    const { canceled, dirPath } = await showSaveMultiDialog();
+    console.log(dirPath);
+    if (canceled || !dirPath) {
+      return;
+    }
+
+    const { readAttachmentData, saveAttachmentToDisk } =
+      window.Signal.Migrations;
+
+    let fullPath;
+    for (const attachment of attachments) {
+      fullPath = await Attachment.save({
+        attachment,
+        index: index + 1,
+        readAttachmentData,
+        saveAttachmentToDisk,
+        timestamp,
+        baseDir: dirPath,
+      });
+    }
+
+    // TODO shouldn't we throw some kind of error here? Original code also doesn't do anything, but...
+    if (fullPath == null) return;
```

Comment:
In `saveAttachments()` (so the function to download a single attachment), I guess this was a check to skip showing the toast, when the user cancels the download/ file picker dialog.
Since I'm doing this check a few lines above, we probably should either remove this check here completely or throw an error, because something went wrong in the `Attachments.save()` method?

Reply:
Yeah, this should probably be a `throw`

Reply:
Changed that, please check the error message.